Lib/test/test_shutil.py

```diff
@@ -84,6 +111,37 @@ def rlistdir(path):
             res.append(name)
     return res
 
+def supports_file2file_sendfile():
```

Comment:
Turning this into a build time configure check is essentially a matter of converting it to a small C program, and then adding it to configure.ac.

https://github.com/python/cpython/blob/master/configure.ac#L4414 is a reasonable example of a comparable check.

Reply:
I see your point. This is not exactly easy though as sendfile() is quite messy.
Both its signature and the inclusion of the right header files is platform dependent. See: https://github.com/giampaolo/pysendfile/blob/master/sendfilemodule.c
My concern is that we would end up duplicating a lot of logic from Modules/posixmodule.c

Reply:
Yeah, given that we need the runtime check no matter what (thanks to the Python-built-against-a-newer-kernel-running-on-an-older-kernel problem), deferring any additional build time checks to a follow up patch makes sense (it also means we can be more confident that the runtime check is robust, since the FreeBSD buildbots will check it).

And given the logic tweak to swap completely back to the Python level implementation when we get `ENOTSOCK`, then the runtime overhead should be negligible (and if it turns out that it isn't, that's when we can reconsider the configure time check option).